lib/test_result.dart

```diff
@@ -75,6 +76,88 @@ class TestResult {
     return TestAttachment.fromJson(response);
   }
 
+  static Future<TestResults> getTestResults({
+    required int testId,
+    int? limit,
+    int? offset,
+  }) async {
+    final url = '/get_results/$testId';
+
+    final queryParameters = <String, dynamic>{
+      'limit': limit,
+      'offset': offset,
+    };
+
+    return _performRequest(
+      requestUrl: url,
+      queryParameters: queryParameters,
+    );
+  }
+
+  static Future<TestResults> getCaseResults({
+    required int runId,
+    required int caseId,
+    String? defectsFilter,
+    int? limit,
+    int? offset,
+    Iterable<int>? statusId,
+  }) async {
+    final url = '/get_results_for_case/$runId/$caseId';
+
+    final queryParameters = <String, dynamic>{
+      'defects_filter': defectsFilter,
+      'limit': limit,
+      'offset': offset,
+      'status_id': statusId?.join(',')
+    };
+
+    return _performRequest(
+      requestUrl: url,
+      queryParameters: queryParameters,
+    );
+  }
+
+  static Future<TestResults> getRunResults({
+    required int runId,
+    DateTime? createdAfter,
+    DateTime? createdBefore,
+    Iterable<int>? createdBy,
+    String? defectsFilter,
+    int? limit,
+    int? offset,
+    Iterable<int>? statusId,
+  }) async {
+    final url = '/get_results_for_run/$runId';
+
+    final queryParameters = <String, dynamic>{
+      'created_after': createdAfter?.millisecondsSinceEpoch,
+      'created_before': createdBefore?.millisecondsSinceEpoch,
+      'created_by': createdBy,
+      'defects_filter': defectsFilter,
+      'limit': limit,
+      'offset': offset,
+      'status_id': statusId?.join(',')
+    };
+
+    return _performRequest(
+      requestUrl: url,
+      queryParameters: queryParameters,
+    );
+  }
+
+  static Future<TestResults> _performRequest({
+    required String requestUrl,
+    dynamic queryParameters,
```

Comment:
`dynamic` should be avoided whenever possible. 
```suggestion
    Map<String, dynamic> queryParameters,
```